internal/pkg/consensus/expected_test.go

```diff
@@ -243,9 +243,30 @@ func TestExpected_RunStateTransition_validateMining(t *testing.T) {
 
 		emptyBLSMessages, emptyMessages, emptyReceipts := emptyMessagesAndReceipts(len(blocks))
 
-		_, err = exp.RunStateTransition(ctx, tipSet, emptyBLSMessages, emptyMessages, emptyReceipts, []block.TipSet{pTipSet}, 0, blocks[0].StateRoot)
+		_, err = exp.RunStateTransition(ctx, tipSet, emptyBLSMessages, emptyMessages, emptyReceipts, []block.TipSet{pTipSet}, uint64(blocks[0].ParentWeight), blocks[0].StateRoot)
 		assert.EqualError(t, err, "block signature invalid")
 	})
+
+	t.Run("returns nil + error when parent weight invalid", func(t *testing.T) {
+		pTipSet := th.RequireNewTipSet(t, genesisBlock)
+		stateTree, err := state.LoadStateTree(ctx, cistore, genesisBlock.StateRoot)
+		require.NoError(t, err)
+		vms := vm.NewStorageMap(bstore)
+
+		blocks, minerToWorker := requireMakeBlocks(ctx, t, pTipSet, stateTree, vms)
+		// Give block 0 an invalid signature
+		blocks[0].BlockSig = blocks[1].BlockSig
+
+		tipSet := th.RequireNewTipSet(t, blocks...)
+		as := consensus.NewFakeActorStateStore(minerPower, totalPower, minerToWorker)
+
+		exp := consensus.NewExpected(cistore, bstore, th.NewFakeProcessor(), th.NewFakeBlockValidator(), as, genesisBlock.Cid(), th.BlockTimeTest, &consensus.FakeElectionMachine{}, &consensus.FakeTicketMachine{})
+
+		emptyBLSMessages, emptyMessages, emptyReceipts := emptyMessagesAndReceipts(len(blocks))
+
+		_, err = exp.RunStateTransition(ctx, tipSet, emptyBLSMessages, emptyMessages, emptyReceipts, []block.TipSet{pTipSet}, 6, blocks[0].StateRoot)
```

Comment:
Is the `6` what makes the parent weight invalid? Please clarify how this test is working.